xtos/include/sof/lib/perf_cnt.h

```diff
@@ -58,7 +76,7 @@ struct perf_cnt_data {
 	#ifdef CONFIG_TIMING_FUNCTIONS
 		#define perf_cnt_get_cpu_ts arch_timing_counter_get
 	#else
-		#define perf_cnt_get_cpu_ts sof_cycle_get_64
+		#define perf_cnt_get_cpu_ts arch_timing_counter_get_DSP
 	#endif	/* CONFIG_TIMING_FUNCTIONS */
```

Comment:
@btian1 I think would be easier to enable CONFIG_TIMING_FUNCTIONS=y and then arch_timing_counter_get() will return CCOUNT on xtensa. No need to have hw specific code on SOF side.

Reply:
@kv2019i @btian1 "DSP clock changes are only done at very specific times with current SOF code (at stream start/stop, not during steady state of audio streaming)"

In reference FW, IPC task and DP processing task can preempt each other (IPC task has budget, for some time has high priority, then switch to low priority). So potentially DP task can start with lower DSP clock, then it can be preempted by IPC task which starts another pipeline and switch DSP clock. Then DP task finish chunk processing with changed clock.
I'm not sure how is it working in SOF?

Moreover, we still have plan (don't we?) to implement DP modules which switch DSP clock in runtime, like WoV or AEC.

Also, in the past platforms, FW had to do HW-specific implementation that switches clocks in runtime, e.g. every time when accessing IMR. Fortunately current platforms do not require this, but who knows...

So in my opinion keeping WLCK as a reference is a safer solution.

Reply:
1. this CONFIG_TIMING_FUNCTIONS only enabled in performance build(with -o app/perf_overlay.conf option).
2. clock measure only happens on module start and module end in function comp_copy(https://github.com/thesofproject/sof/blob/main/src/audio/component.c)
    is it possible clock will switch inside module?

Reply:
OK, it it is only enabled in performance build, then I'm completely fine with this.